build-system/pr-check/build-targets.js

```diff
@@ -37,6 +41,16 @@ function isOwnersFile(file) {
   return file.endsWith('OWNERS');
 }
 
+/**
+ * Checks if the given file is the renovate config file.
+ *
+ * @param {string} file
+ * @return {boolean}
+ */
+function isRenovateConfigFile(file) {
+  return RENOVATE_CONFIG_REGEX.test(file);
```

Comment:
Other build targets in this file use a simple equality check. Why not do the same here?

```js
return file == `.renovaterc.json`;
```

Reply:
If down the line we were to rename the file, such as to use JSON5 to allow for including comments, this would silently ignore/not catch the renamed configuration file. I was thinking it makes sense to catch anything that renovate will automatically consider a configuration file

Reply:
IMO, changing the extension of a config file is a deliberate action that ought make sure that the build target is still accurate. For example, `BABEL_PLUGIN` and `PACKAGE_UPGRADE` will have to change if we ever change the file extensions for babel configurations or package versions.

Reply:
That need for updating the build system in order to change a config file strikes me as more of a bug than a feature. I'm guessing your main concern is clarity, specifically that the regex isn't trivial to parse. How would you feel if I instead defined a `new Set([...])` that listed the options explicity, and made this `RENOVATE_CONFIG_FILES.has(file)`?

Reply:
Yes, my main concerns are clarity and simplicity of code. The regex you have right now is hard to read and cannot be found when someone searches the codebase for the name of the current config file. If you convert it to a list, it can be found, but it's still not obvious from the code which file is actually in use.

Reply:
Why do you feel it needs to be obvious from the code which file is in use? Any IDE (or command-line) search for "renovate" turns up two files: `check-renovate-config.js` and `.renovaterc.json`. I'm not sure I understand the benefits of tying the implementation to a specific filename.

With regards to brittle-ness, a list of known options is going to be less brittle than a single sample string. We could import the list directly `const {configFileNames} = require('renovate/dist/config/app-strings');`, which would ensure freshness in the rare case that the name does change (possibly adding a comment for grep-ability). Over the past several years, the only changes have been additive (adding GitLab and JSON5 support). The benefit of a hardcoded list is that we can include config files which we know will trigger the build target, even if not explicitly listed. For example, I just very recently discovered that if `amphtml/config.js` exists, Renovate will run and validate it. It's reasonable to believe someone down the line might add a `config.js` to the repo, and it would go undetected, silently breaking Renovate in the process.

Ultimately, I feel checking against a single filename is itself very brittle. At the very least, I'd lean toward setting that build target on any filename containing "renovate" or some similar, wider check.

Reply:
The root cause of my objection to this approach is that it deviates from the approach used by every other target matcher in this file. All of them are based on the current set filenames in the `amphtml` repo, as they stand today, even though they could have been written to accommodate all valid possibilities.

You did raise a good higher-level point about wanting the build-target system to be more flexible. We could set out to make it so, but I think we should do it across the system, rather than use different approaches in different parts. (Also, I think such a refactor is low on our current list of priorities.)

Hopefully, my other comment (where I realized we don't actually need `isRenovateConfigFile()`) makes this much less of an issue. Let me know what you think.